Client/Frontend/Browser/TabManager.swift

```diff
@@ -249,7 +249,9 @@ extension TabManager {
         }
 
         let selectedIndex: Int = coder.decodeIntegerForKey("selectedIndex")
-        self.selectTab(self.tabs[selectedIndex])
+        if (selectedIndex >= 0) {
```

Comment:
Can you please remove the ellipsis? And I think it is a good idea to also check if the selectedIndex is not greater than the number of tabs. (Be aware that the selectedIndex is zero-based). If the selectedIndex is not in range then just open the first tab. If there is one.